File: backend/DTO/RecordDTO.cs
using System.Text.Json.Serialization;
using backend.Models;

namespace backend.DTO;

public record CreateRecordDTO
{
    [JsonPropertyName("course_code")]
    public required string CourseCode { get; init; }
    [JsonPropertyName("course_id")]
    public required string CourseID { get; init; }
    [JsonPropertyName("course")]
    public required string Course { get; init; }
    [JsonPropertyName("course_icon")]
    public string? CourseIcon { get; init; }
    [JsonPropertyName("assignment_code")]
    public required string AssignmentCode { get; init; }
    [JsonPropertyName("assignment")]
    public required string Assignment { get; init; }
    [JsonPropertyName("problems")]
    public required List<Problem> Problems { get; init; }
}

public record RecordDTO
{
    [JsonPropertyName("id")]
    public string? ID { get; init; }
    [JsonPropertyName("assignment_code")]
    public required string AssignmentCode { get; init; }
    [JsonPropertyName("problems")]
    public required List<Problem> Problems { get; init; }
    [JsonPropertyName("created_at")]
    public DateTime? CreatedAt { get; init; }
}

File: frontend/src/api/dto/record.dto.ts
import { Problem, Record } from "@/src/types";

export type RecordDTO = {
  id: string;
  assignment_code: string;
  code: string;
  problems: Problem[];
  created_at: string;
};

export const parseRecordDTO = (data: RecordDTO): Record => ({
  id: data.id,
  assignmentCode: data.assignment_code,
  problems: data.problems,
  createdAt: data.created_at,
});

export const parseRecordDTOList = (data: RecordDTO[]): Record[] => {
  return data.map(parseRecordDTO);
};
